File: app/src/main/java/de/christinecoenen/code/zapp/ChangelogActivity.java
package de.christinecoenen.code.zapp;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;

import com.github.porokoro.paperboy.ViewTypes;
import com.github.porokoro.paperboy.builders.PaperboyChainBuilder;

public class ChangelogActivity extends AppCompatActivity {

	public static Intent getStartIntent(Context context) {
		return new Intent(context, ChangelogActivity.class);
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_changelog);

		ActionBar actionBar = getSupportActionBar();
		if (actionBar != null) {
			actionBar.setHomeButtonEnabled(true);
			actionBar.setDisplayHomeAsUpEnabled(true);
		}

		if (savedInstanceState == null) {
			Fragment fragment = new PaperboyChainBuilder(this)
					.viewType(ViewTypes.HEADER)
					.buildFragment();

			getSupportFragmentManager()
					.beginTransaction()
					.add(R.id.activity_changelog, fragment)
					.commit();
		}
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/tv/changelog/ChangelogActivity.kt
package de.christinecoenen.code.zapp.tv.changelog

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import de.christinecoenen.code.zapp.R
import de.christinecoenen.code.zapp.databinding.TvActivityChangelogBinding
import de.christinecoenen.code.zapp.utils.io.IoUtils.readAllText
import de.christinecoenen.code.zapp.utils.system.IStartableActivity
import ru.noties.markwon.Markwon

class ChangelogActivity : Activity() {

	companion object : IStartableActivity {
		override fun getStartIntent(context: Context?): Intent =
			Intent(context, ChangelogActivity::class.java)
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		val binding = TvActivityChangelogBinding.inflate(layoutInflater)

		setContentView(binding.root)

		val markdown = resources.readAllText(R.raw.changelog)
		Markwon.setMarkdown(binding.txtChangelog, markdown)
	}
}
